### PR TITLE
Fix _refs.lcm using floating point maths

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80906
* #80905
* __->__ #82950

`lcm` is meant to use integer maths, but the use of `true_divide`
introduces a promotion to float and thus a loss of precision.

This also introduces promoting low precision integers to int32 which
is required for 100% consistency with the C++ implementation since the
"usual arithmetic conversions" means the intermediate terms are
calculated to `int` precision in C++. This only really matters when the
lower precision dtype would overflow, however the test cases for lcm
do involve overflows.

### CODE REVIEW

**Comment by peterbell10** [comment]
@ngimel ping

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -1276,8 +1276,17 @@ def isclose(
...
+    dtype = a.dtype
+    promote_to_int = dtype in (torch.int8, torch.int16)
+    if promote_to_int:
+        a = prims.convert_element_type(a, torch.int32)
+        b = prims.convert_element_type(b, torch.int32)
+
+    g = torch.gcd(a, b)
+    # Avoid division by zero in case gcd(0, 0) == 0
+    g = torch.where(g == 0, 1, g)
+    res = torch.abs(prims.div(a, g) * b)
```
Comment (ngimel) [inline torch/_refs/__init__.py:1288]: is it possible for `a` and `g` to have different types (e.g. a is int32, and b is int64, which probably will result in g int64). `prim.div` won't work for this case?

Comment (peterbell10) [reply torch/_refs/__init__.py:1288]: Both operators call `_make_elementwise_binary_reference` with the same `type_promotion_kind` so I don't think that should ever happen.

**Review by ngimel — APPROVED** [review]
